### DISCUSSION TITLE
Defaulting to static rendering even with data mutations. What I am missing?

### DISCUSSION THREAD

**Discussion by peibolsang** [vercel/next.js#62578] [App Router]
I have been experimenting with ALL Next.js 14 rendering options and [I have documented my findings](https://github.com/peibolsang/octotype/issues/12#issuecomment-1960549044).

All the findings above prompt the following question: why not mark all our pages (landing and dynamic routes) as static pages to be served from the CDN with `export const dynamic="force static"` to get the optimum performance? 

Then, we can revalidate tags/pages when data mutations happen.
1. If data mutations happen outside our system, we can expose a Webhook that forces the revalidation through `revalidatePath()` or `revalidateTag()` so users will get fresh data after a mutation.
2. If data mutations happen in our system, we can implement server actions that force the revalidation, too, so users will get fresh data after a mutation.

I tested this, and it works. For use case number 2 (i.e., data mutations happen in your system), the data revalidation itself is fast, but:
- If the mutation was provoked by another user, the static page would be served with an `x-vercel-cache=REVALIDATED` the next time it is accessed. The first access will be slow, but subsequent access will get a `HIT` again and will be faster.
- If the same user provoked the mutation, seeing the fresh data on screen takes a while because the page has to be compiled and served again entirely after revalidation. Then, the same thing happens: the page would be served with an `x-vercel-cache=REVALIDATED` the next time it is accessed. The first access will be slow, but subsequent access will get a `HIT` again and will be faster.

What am I missing? Is this a good default approach?

> Note: The only disadvantage of `force-static` is that it does not work with the `edge` because this issue has not been solved: https://github.com/vercel/next.js/issues/42112

**Comment by markoleavy** [comment]
> why not mark all our pages (landing and dynamic routes) as static pages to be served from the CDN with export const dynamic="force static" to get the optimum performance?

As far as I know dynamic="force static" works for dynamic routes only. Render strategy for static routes will be determined base on the cache strategies. 
You cold chose to disable static rendering for pages that request on demand server rendering (i.e. personal pages) and to speed up large builds.
